File: server/internal/models/terminal_type.go
package models

import "time"

// TerminalType 终端类型
type TerminalType struct {
	ID          int64     `json:"id" gorm:"primaryKey"`
	ChannelID   int64     `json:"channel_id" gorm:"not null;index"`
	ChannelCode string    `json:"channel_code" gorm:"type:varchar(50);not null;index"`
	BrandCode   string    `json:"brand_code" gorm:"type:varchar(50);not null"`
	BrandName   string    `json:"brand_name" gorm:"type:varchar(100);not null"`
	ModelCode   string    `json:"model_code" gorm:"type:varchar(50);not null"`
	ModelName   string    `json:"model_name" gorm:"type:varchar(100)"`
	Description string    `json:"description" gorm:"type:text"`
	Status      int16     `json:"status" gorm:"default:1"` // 1:启用 0:禁用
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`

	// 关联
	Channel *Channel `json:"channel,omitempty" gorm:"foreignKey:ChannelID"`
}

// TableName 表名
func (TerminalType) TableName() string {
	return "terminal_types"
}

// TerminalTypeStatus 终端类型状态
const (
	TerminalTypeStatusDisabled = 0 // 禁用
	TerminalTypeStatusEnabled  = 1 // 启用
)

// IsEnabled 是否启用
func (t *TerminalType) IsEnabled() bool {
	return t.Status == TerminalTypeStatusEnabled
}

// FullName 完整名称（品牌 - 型号）
func (t *TerminalType) FullName() string {
	if t.ModelName != "" {
		return t.BrandName + " - " + t.ModelName
	}
	return t.BrandName + " - " + t.ModelCode
}

File: web/src/types/terminalType.ts
// 终端类型
export interface TerminalType {
  id: number
  channel_id: number
  channel_code: string
  channel_name: string
  brand_code: string
  brand_name: string
  model_code: string
  model_name: string
  full_name: string
  description: string
  status: number
  created_at: string
  updated_at: string
}

// 终端类型查询参数
export interface TerminalTypeQueryParams {
  channel_id?: number
  status?: number
  keyword?: string
  page?: number
  page_size?: number
}

// 创建终端类型参数
export interface CreateTerminalTypeParams {
  channel_id: number
  brand_code: string
  brand_name: string
  model_code: string
  model_name?: string
  description?: string
}

// 更新终端类型参数
export interface UpdateTerminalTypeParams {
  brand_code?: string
  brand_name?: string
  model_code?: string
  model_name?: string
  description?: string
}

// 终端类型状态
export const TerminalTypeStatus = {
  DISABLED: 0,
  ENABLED: 1,
} as const

export const TerminalTypeStatusText: Record<number, string> = {
  [TerminalTypeStatus.DISABLED]: '禁用',
  [TerminalTypeStatus.ENABLED]: '启用',
}
